llvm/test/Transforms/InstCombine/select.ll

```diff
@@ -2683,7 +2683,8 @@ define void @cond_freeze_multipleuses(i8 %x, i8 %y) {
 
 define i32 @select_freeze_icmp_eq(i32 %x, i32 %y) {
```

Comment:
This comes from: https://github.com/llvm/llvm-project/blob/42350f428db0d053610a82a747eb240afc4d9250/llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp#L2845

We regress this case by inserting an unnecessary freeze now, but I don't think this is important. We can drop the specialized transform after this patch.